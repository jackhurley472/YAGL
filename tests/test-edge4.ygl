Graph G;

Node n("Adam");
Node n2("James");
Node n3("Tank");
Node n4("Jack");

G + n;
G + n2;
G + n3;
G + n4;

int b = 3;

/* All different types of edges */

G: n -> n2; 		/* Default 1 */
G: n ->2 n3; 		/* Simply just a literal */
G: n2 ->b n3; 		/* Simply just a variable */
G: n3 ->|b + 1| n2;	/* Expression */
G: n3 ->|5| n4;		/* Optional | */

printGraph(G);


