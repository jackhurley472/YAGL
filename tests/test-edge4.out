============== Graph Print ===============
	Stats: 	4	4	3

Nodes:
Node (0) : Adam --- Node (1) : James --- Node (2) : Tank
Node (3) : Jack

Edges:
[(0) : Adam] ---(1)---> [(1) : James]
[(0) : Adam] ---(2)---> [(2) : Tank]
[(1) : James] ---(3)---> [(2) : Tank]
[(2) : Tank] ---(4)---> [(1) : James]
[(2) : Tank] ---(5)---> [(3) : Jack]
============ End Graph Print =============
